Xamarin.Forms.Core/BindableLayout.cs

```diff
@@ -223,9 +223,9 @@ void ItemsSourceCollectionChanged(object sender, NotifyCollectionChangedEventArg
 					{
 						if (e.OldStartingIndex == -1)
 							goto case NotifyCollectionChangedAction.Reset;
-						for (int i = 0; i < e.OldItems.Count; ++i)
+						for (int i = 0; i < e.OldItems.Count; i++)
 						{
```

Comment:
I wanted to fix it today, but thanks for your help.
I realized my mistake and a better fix I think is to actually remove the `for` iteration and replace with a `while`, using the index and increment it is not necessary:
`while(e.OldItems.Count > 0)
     layout.Children.RemoveAt(0);

Reply:
Also, the unit test must be added